File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int index = round(0.0588 * (float)letters / (float)words * 100.0 - 0.296 * (float)sentences / (float)words * 100.0 - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Take string and return # of letters
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        // If char is letter
        char c = tolower(text[i]);
        if (c >= 97 && c <= 122)
        {
            ++letters;
        }
    }

    return letters;
}

// Take string and return # of words
int count_words(string text)
{
    int words = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        // If char is space
        if (text[i] == 32)
        {
            ++words;
        }
    }

    return words;
}

// Take string and return # of sentences (# of .?!)
int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        // If char is period, !, or ?
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            ++sentences;
        }
    }

    return sentences;
}

File: sentimental-readability/readability.py
# c stands for char

import cs50


def main():
    text = cs50.get_string("Text: ")

    # Count these values
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Coleman-Liau Index
    index = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8)

    # Decide grade level
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    letters = 0

    for c in text:
        uni_c = ord(c.lower())
        # If char is letter
        if uni_c >= 97 and uni_c <= 122:
            letters += 1

    return letters


def count_words(text):
    words = 1

    for c in text:
        uni_c = ord(c)
        # If char is space
        if uni_c == 32:
            words += 1

    return words


def count_sentences(text):
    sentences = 0

    for c in text:
        uni_c = ord(c)
        # If char is period, !, or ?
        if uni_c == 46 or uni_c == 33 or uni_c == 63:
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()